### PR TITLE
Use more expressive object macro syntax for cross-crate inheritance

### PR DESCRIPTION
**Pull request by gkoz**
This implementation is hacky but should serve at least as a proof of concept.
https://github.com/gtk-rs/glib/pull/118

### CODE REVIEW

File: Gir_Gio.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[options]
+girs_dir = "gir-files"
+library = "Gio"
+version = "2.0"
+min_cfg_version = "2.32"
+target_path = "."
```
Comment (EPashkin) [inline Gir_Gio.toml]: #target_path = "./gio" plz

File: Gir_Gio.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[options]
+girs_dir = "gir-files"
```
Comment (EPashkin) [inline Gir_Gio.toml]: girs_dir = "../gir-files"

**Comment by gkoz** [comment]
Looks like `git2` trips an LLVM assertion on current nightly, we can ignore that.

File: Gir_Gio.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "Gio.MenuAttributeIter",
+    "Gio.MenuItem",
+    "Gio.MenuLinkIter",
+    "Gio.MenuModel",
+    "Gio.Permission",
+    "Gio.Settings",
+    "Gio.SimpleAction",
+    "Gio.SimpleActionGroup",
+    "Gio.SimplePermission",
+]
```
Comment (EPashkin) [inline Gir_Gio.toml]: Not sure if it works on Windows but
    "Gio.Drive",
    "Gio.Mount",
    "Gio.Volume",
    "Gio.VolumeMonitor",

File: Gir_Gio.toml (hunk 1 above)
Comment (EPashkin) [inline Gir_Gio.toml]: And `/gio` in `.gitignore`

File: Gir_Gio.toml (hunk 3 above)
Comment (gkoz) [inline Gir_Gio.toml]: At least they don't seem to require `gio-unix` so we could try them too. There are some thread safety issues though:

> GVolumeMonitor is not [thread-default-context aware](https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#g-main-context-push-thread-default), and so should not be used other than from the main thread, with no thread-default-context active.

We don't have a strategy for dealing with that outside `gdk`/`gtk`. Could consider lifting the main_thread stuff to `glib` but here the limitations are documented in a cryptic way :(

**Comment by gkoz** [comment]
Updated.

**Comment by gkoz** [comment]
Now we don't add the `[]` if the simpler parents syntax is sufficient, and format the extended syntax better.

**Comment by EPashkin** [comment]
That much better, thanks.

**Comment by EPashkin** [comment]
Sorry, not seen that it ready for merging.

**Pull request merged**